src/lib/cli_lib/arg_type.ml

```diff
@@ -12,17 +12,34 @@ let int16 =
 
 let public_key_compressed =
   Command.Arg_type.create (fun s ->
-      try Public_key.Compressed.of_base58_check_exn s
-      with e ->
+      let error_string e =
         let random = Public_key.compress (Keypair.create ()).public_key in
         eprintf
           "Error parsing command line.  Run with -help for usage information.\n\n\
-           Couldn't read public key (Invalid key format)\n\
+           Couldn't read public key\n\
           \ %s\n\
           \ - here's a sample one: %s\n"
-          (Error.to_string_hum (Error.of_exn e))
+          (Error.to_string_hum e)
           (Public_key.Compressed.to_base58_check random) ;
-        exit 1 )
+        exit 1
+      in
+      let key =
+        try Ok (Public_key.Compressed.of_base58_check_exn s)
```

Comment:
I think it might make more sense here to do `Public_key.Compressed.of_base58_check` and match on the option. We don't need to surface the `Option.value_exn` error.